### PR TITLE
Convert ac commands to rzshell

### PR DESCRIPTION
**Pull request by p371jah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Printing APIs `rz_analysis_class_print()`, `rz_analysis_class_json()`, `rz_analysis_class_list()`, `rz_analysis_class_list_bases()`, `rz_analysis_class_list_vtables()`, `rz_analysis_class_list_vtable_offset_functions()` were removed from `rz-analysis`; they are part of commands handlers now.
- `rz_analysis_class_create()` was refactored to return `RzAnalysisClassErr`
- `acl <class>` was replaced with `aci`
- `acb <class>` was replaced with `acbl`
- `acv <class>` was replaced with `acvl`

**Test plan**

...

**Closing issues**

Related: #1442

### CODE REVIEW

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -6601,3 +6601,294 @@ RZ_API void rz_core_analysis_cc_init(RzCore *core) {
...
+	ListJsonCtx ctx;
+	ctx.analysis = analysis;
+	ctx.pj = pj;
+	pj_a(pj);
+	rz_analysis_class_foreach(analysis, rz_core_analysis_class_list_json_cb, &ctx);
+	pj_end(pj);
+	return;
+}
+
+RZ_API void rz_core_analysis_class_list(RzAnalysis *analysis, RzCmdStateOutput *state) {
```
Comment (XVilka) [inline librz/core/canalysis.c]: In general, we try to follow the naming when if function ends with `_list` it should return `RzList`, for printing it's better to use the suffix `_print` instead.

File: librz/core/canalysis.c [hunk 2]
```diff
@@ -6601,3 +6601,294 @@ RZ_API void rz_core_analysis_cc_init(RzCore *core) {
...
+		}
+	} else {
+		ls_foreach (classes, iter, kv) {
+			rz_core_analysis_class_print(analysis, sdbkv_key(kv), state->mode == RZ_OUTPUT_MODE_LONG);
+		}
+	}
+	ls_free(classes);
+}
+
+RZ_API void rz_core_analysis_class_list_bases(RzAnalysis *analysis, const char *class_name) {
```
Comment (XVilka) [inline librz/core/canalysis.c]: Same applies to all these functions. Moreover, I wonder if it is better to make them `RZ_IPI` instead, they don't make sense as the API for the use outside of Rizin core.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/canalysis.c (hunk 2 above)
Comment (p371jah) [reply librz/core/canalysis.c]: Yep it was on my list.

**Review by XVilka — APPROVED** [review]
LGTM. @thestr4ng3r please take a look since you are the one who implemented these commands in the first place.

File: librz/core/cmd_descs/cmd_analysis.yaml [hunk 3]
```diff
@@ -1319,3 +1319,155 @@ commands:
...
             type: RZ_CMD_ARG_TYPE_RZNUM
+  - name: ac
+    summary: Classes
+    subcommands:
+      - name: ac
+        summary: Add class
+        cname: analysis_class_add
+        args:
+          - name: class_name
+            type: RZ_CMD_ARG_TYPE_STRING
```
Comment (ret2libc) [inline librz/core/cmd_descs/cmd_analysis.yaml:1312]: shall we add a new arg type for this?

**Review by ret2libc — COMMENTED** [review]
LGTM

File: librz/core/cmd_descs/cmd_analysis.yaml (hunk 3 above)
Comment (p371jah) [reply librz/core/cmd_descs/cmd_analysis.yaml:1312]: For autocompletion?

Comment (wargio) [reply librz/core/cmd_descs/cmd_analysis.yaml:1312]: i think it is a good idea to have a new type

Comment (ret2libc) [reply librz/core/cmd_descs/cmd_analysis.yaml:1312]: yes, for future autocompletion.

Comment (XVilka) [reply librz/core/cmd_descs/cmd_analysis.yaml:1312]: Can be done in a separate PR, merging this.

**Pull request merged**